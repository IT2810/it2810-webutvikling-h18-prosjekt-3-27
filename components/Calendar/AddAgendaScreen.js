import React, {Component} from "react";
import {
  View,
  Text,
  StyleSheet,
  TextInput,
  Button
} from "react-native";

class AddAgendaScreen extends Component {
  static navigationOptions = {
    title: "Create event"
  };

  state = {
    name: "",
    note: ""
  };

  handleNameChange = (text) => {
    this.setState({name: text});
  };

  handleNoteChange = (text) => {
    this.setState({note: text});
  };

  handleAddAgendaPress = () => {
    // button for adding a new agenda was clicked
    const callbackAddAgenda = this.props.navigation.getParam("addAgenda");
    if (!callbackAddAgenda)
      return;
    // create object for the new agenda
    const agenda = {
      date: this.props.navigation.getParam("date"),
      name: this.state.name,
      note: this.state.note
    };
    // pass it back to the calendar screen
    callbackAddAgenda(agenda);
    // then navigate back
    this.props.navigation.navigate("Calendar");
  };

  render() {
    return (
      <View style={styles.container}>
        <Text>Name:</Text>
        <TextInput
          style={styles.textInput}
          value={this.state.name}
          defaultValue={"Name"}
          onChangeText={this.handleNameChange}
        />
        <Text>Description:</Text>
        <TextInput
          style={styles.textInput}
          value={this.state.note}
          defaultValue={"A short description..."}
          onChangeText={this.handleNoteChange}
        />
        <Button title={"Add new event"} onPress={this.handleAddAgendaPress}/>
      </View>
    );
  }
}

export default AddAgendaScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "flex-start"
  },
  textInput: {
    width: "80%"
  }
});
