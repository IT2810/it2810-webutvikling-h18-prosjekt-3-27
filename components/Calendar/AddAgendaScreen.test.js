import React from "react";
import renderer from "react-test-renderer";
import AddAgendaScreen from "./AddAgendaScreen";

it("renders correctly", () => {
  const tree = renderer
    .create(<AddAgendaScreen/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
