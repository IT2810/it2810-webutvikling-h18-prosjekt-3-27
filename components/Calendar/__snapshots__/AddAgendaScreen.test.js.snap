// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "marginBottom": 10,
          "marginTop": 10,
        },
        Object {
          "fontSize": 26,
        },
      ]
    }
  >
    No date
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginBottom": 10,
        "marginTop": 10,
      }
    }
  >
    Name:
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Name"
    style={
      Array [
        Object {
          "marginBottom": 10,
          "marginTop": 10,
        },
        Object {
          "borderColor": "#000",
          "borderWidth": 1,
          "padding": 5,
          "width": "80%",
        },
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginBottom": 10,
        "marginTop": 10,
      }
    }
  >
    Description:
  </Text>
  <TextInput
    allowFontScaling={true}
    multiline={true}
    numberOfLines={5}
    onChangeText={[Function]}
    placeholder="Enter description..."
    style={
      Array [
        Object {
          "marginBottom": 10,
          "marginTop": 10,
        },
        Object {
          "borderColor": "#000",
          "borderWidth": 1,
          "padding": 5,
          "width": "80%",
        },
      ]
    }
    textAlignVertical="top"
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Add new event
      </Text>
    </View>
  </View>
</View>
`;
